### PR TITLE
Push xDS when SE TargetPort is changed

### PR DESCRIPTION
**Pull request by dddddai**
**Please provide a description of this PR:**
Fixes https://github.com/istio/istio/issues/49878

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 1]
```diff
@@ -1168,10 +1168,10 @@ func serviceUpdateNeedsPush(prev, curr *v1.Service, preConv, currConv *model.Ser
...
 	// `preConv.Equals(currConv)` already makes sure the length of ports is not changed
 	if prev != nil && curr != nil {
-		for i := 0; i < len(prev.Spec.Ports); i++ {
-			if prev.Spec.Ports[i].TargetPort != curr.Spec.Ports[i].TargetPort {
-				return true
-			}
+		if !slices.EqualFunc(prev.Spec.Ports, curr.Spec.Ports, func(a, b v1.ServicePort) bool {
+			return a.TargetPort == b.TargetPort
+		}) {
+			return true
```
Comment (dddddai) [inline pilot/pkg/serviceregistry/kube/controller/controller.go:1174]: This is not related. Just to make it aligned with the code in SE controller

**Review by hzxuzhonghu — COMMENTED** [review]
can we do eds push when targetport changed

**Comment by dddddai** [comment]
> can we do eds push when targetport changed

TargetPort also impacts LDS(see https://github.com/istio/istio/issues/48580). I believe we need a full push

**Review by howardjohn — APPROVED** [review]
